sage.py

```diff
@@ -150,74 +136,28 @@ def _parse(unparsed_data, filter_alerts=False):
-        remaining[MicroAttackStage.NON_MALICIOUS.value] = 0  # mcat that has been filtered (non-malicious)
-
-    # Use ordered dictionaries to make sure that the labels (categories) are aligned
-    b1 = OrderedDict(sorted(original.items()))
-    b2 = OrderedDict(sorted(remaining.items()))
-
-    plt.figure(figsize=(20, 20))
-    plt.gcf().subplots_adjust(bottom=0.2)  # To fit the x-labels
-
-    # Set width and height of bar
-    bar_width = 0.4
-    bars1 = [x for x in b1.values()]
-    bars2 = [x for x in b2.values()]
-
-    # Set position of bar on x-axis
-    r1 = np.arange(len(bars1))
-    r2 = [x + bar_width for x in r1]
-
-    # Make the plot
-    plt.bar(r1, bars1, color='skyblue', width=bar_width, edgecolor='white', label='Raw')
-    plt.bar(r2, bars2, color='salmon', width=bar_width, edgecolor='white', label='Cleaned')
-
-    labels = [micro[x].split('.')[1] for x in b1.keys()]
-
-    # Add xticks in the middle of the group bars
-    plt.ylabel('Frequency', fontweight='bold', fontsize='20')
-    plt.xlabel('Alert categories', fontweight='bold', fontsize='20')
-    plt.xticks([(x + bar_width / 2) for x in r1], labels, fontsize='10', rotation='vertical')
-    plt.yticks(fontsize='20')
-    plt.title('High-frequency Alert Filtering', fontweight='bold', fontsize='20')
-
-    # Create legend & show graphic
-    plt.legend(prop={'size': 20})
-    plt.show()
-    return
-
-
 # Step 1.2: Remove duplicate alerts (defined by the alert_filtering_window parameter)
 def _remove_duplicates(unfiltered_alerts, plot=False, gap=1.0):
     filtered_alerts = [unfiltered_alerts[x] for x in range(1, len(unfiltered_alerts))
-                       if unfiltered_alerts[x][9] != MicroAttackStage.NON_MALICIOUS.value  # Filter out non-malicious alerts
+                       if unfiltered_alerts[x][9] != MicroAttackStage.NON_MALICIOUS.value  # Skip non-malicious alerts
                        and not (unfiltered_alerts[x][0] <= gap  # Diff from previous alert is less than gap sec
                                 and unfiltered_alerts[x][1] == unfiltered_alerts[x - 1][1]  # Same srcIP
                                 and unfiltered_alerts[x][3] == unfiltered_alerts[x - 1][3]  # Same destIP
                                 and unfiltered_alerts[x][5] == unfiltered_alerts[x - 1][5]  # Same suricata category
                                 and unfiltered_alerts[x][2] == unfiltered_alerts[x - 1][2]  # Same srcPort
                                 and unfiltered_alerts[x][4] == unfiltered_alerts[x - 1][4])]  # Same destPort
     if plot:
-        _plot_alert_filtering(unfiltered_alerts, filtered_alerts)
+        plot_alert_filtering(unfiltered_alerts, filtered_alerts)
 
     print('Filtered # alerts (remaining):', len(filtered_alerts))
     return filtered_alerts
 
 
 # Step 1: Read the input alerts
-def load_data(path_to_alerts, filtering_window):
+def load_data(path_to_alerts, filtering_window, start_hour, end_hour):
```

Comment:
The `start_hour` and `end_hour` in this method, even though they are global variables, are a bit confusing when reading the code (in the sense that it is not immediately clear where they come from). Do you think it would be better to pass them as function arguments, like other command-line parameters (for consistency)? They could also be capitalised, but that would be inconsistent...

Reply:
They can be handled just like t and w are. All CMD parameters should be handled the same way